src/test/java/co/cask/hydrator/salesforce/parser/SalesforceQueryParserTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.salesforce.parser;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SalesforceQueryParserTest {
+  private static final Map<String, Object[]> queries = new HashMap<>();
+
+  static {
+    queries.put("SELECT Id, Name FROM Opportunity",
```

Comment:
what happens when the following test is run: "(SELECT Id, Name FROM Opportunity)"? I believe this is a valid sql statement?

Reply:
I agree. But salesforce server SOQL parser does not like your idea :). The bellow exception is a response from Salesforce Bulk API not from my code.

> java.lang.Exception: co.cask.hydrator.salesforce.BulkAPIBatchException: Batch failed. BatchId='7511i000002aj0uAAA', Reason='InvalidBatch : Failed to process query: MALFORMED_QUERY: unexpected token: ('
	at org.apache.hadoop.mapred.LocalJobRunner$Job.runTasks(LocalJobRunner.java:489) ~[org.apache.hadoop.hadoop-mapreduce-client-common-2.8.0.jar:na]
	at org.apache.hadoop.mapred.LocalJobRunner$Job.run(LocalJobRunner.java:549) ~[org.apache.hadoop.hadoop-mapreduce-client-common-2.8.0.jar:na]
Caused by: co.cask.hydrator.salesforce.BulkAPIBatchException: Batch failed. BatchId='7511i000002aj0uAAA', Reason='InvalidBatch : Failed to process query: MALFORMED_QUERY: unexpected token: ('

Reply:
Hmm ok, thanks for testing

Reply:
thanks for reviewing